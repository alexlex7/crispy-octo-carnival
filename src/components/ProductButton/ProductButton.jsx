import React from 'react';
import styles from './ProductButton.module.css';
import PropTypes from 'prop-types';
import { ReactComponent as ShareIcon } from '../../assets/images/share.svg';

export default function ProductButton({ type, onClick }) {
  return (
    <button className={styles.btn} type="button" onClick={onClick}>
      {type === 'share' && <ShareIcon className={styles.icon} />}
    </button>
  );
}

ProductButton.propTypes = {
  type: PropTypes.oneOf(['favorite', 'share']).isRequired,
  onClick: PropTypes.func,
};
