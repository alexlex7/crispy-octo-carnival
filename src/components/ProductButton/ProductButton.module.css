.btn {
  width: 26px;
  height: 26px;
  padding: 0;

  display: flex;
  justify-content: center;
  /* align-items: center; */

  /* background-color: #ef5423; */
  background-color: #ecf4ff;
  border-radius: 100%;

  border: none;
  transition: 0.3s;
}

.btn:hover,
.btn:focus {
  background-color: antiquewhite;
}

.icon {
  display: block;
  align-self: center;
  fill: black;
}
