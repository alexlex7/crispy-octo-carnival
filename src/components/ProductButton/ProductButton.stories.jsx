import React from 'react';

import ProductButton from './ProductButton';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Products/ProductButton',
  component: ProductButton,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <ProductButton {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  type: 'share',
};
