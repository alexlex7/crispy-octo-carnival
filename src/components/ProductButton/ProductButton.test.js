import { render, screen } from '@testing-library/react';
import ProductButton from './ProductButton';

test('render product item', () => {
  const { asFragment } = render(<ProductButton type="share" />);

  const btn = screen.getByRole('button');
  expect(btn).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});
