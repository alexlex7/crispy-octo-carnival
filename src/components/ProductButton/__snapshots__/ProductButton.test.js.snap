// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render product item 1`] = `
<DocumentFragment>
  <button
    class="btn"
    type="button"
  >
    <svg
      class="icon"
    >
      share.svg
    </svg>
  </button>
</DocumentFragment>
`;
